### PR TITLE
Precursor for #25607

### PR DESCRIPTION
**Pull request by impact27**
- [x] precursor for #25607
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
Nothing new

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Can you summarize what's changing here, and the reason for it?

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -328,6 +328,106 @@ def _setitem_with_indexer(self, indexer, value):
                     take_split_path = True
                     break
 
+        def can_do_equal_len(labels, value, plane_indexer, lplane_indexer):
```
Comment (jreback) [inline pandas/core/indexing.py]: can you make these module level functions instead?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@impact27 happy to take straight reorg that passes everywhere. pls merge master. and update to comments.

**Comment by impact27** [comment]
I will take care of that over the next few days, probably this week-end.

File: pandas/core/indexing.py (hunk 1 above)
Comment (impact27) [reply pandas/core/indexing.py]: Well not really, But I have made them methods instead.

**Comment by WillAyd** [comment]
@impact27 do we need this PR or is it a duplicate of the other one you link to?

**Comment by impact27** [comment]
This one doesn’t do anything, it is just code reorganisation. The idea is to not do too many things in one PR.

**Comment by jreback** [comment]
@impact27 if you can merge master and get passing, pls ping

**Comment by jreback** [comment]
pls merge master; ping when passing

**Comment by jreback** [comment]
closing as stale; ping if you want to continue
